corehq/apps/locations/views.py

```diff
@@ -811,3 +813,39 @@ def loc_to_payload(loc):
             locs = locs.filter(name__icontains=query)
 
         return json_response(map(loc_to_payload, locs[:10]))
+
+
+class DowngradeLocationsView(BaseDomainView):
+    """
+    This page takes the place of the location pages if a domain gets
+    downgraded, but still has users assigned to locations.
+    """
+    template_name = 'locations/downgrade_locations.html'
+    urlname = 'downgrade_locations'
+
+    def dispatch(self, *args, **kwargs):
+        if not users_have_locations(self.domain):  # irrelevant, redirect
+            redirect_url = reverse('users_default', args=[self.domain])
+            return HttpResponseRedirect(redirect_url)
```

Comment:
Why would someone end up on this page if they don't have users with locations? Just curious whether this is a shouldn't-ever-happen redirect or if there's somewhere currently sending users here that should be updated, given that redirects aren't free.